### ISSUE TITLE
'400 Bad Request' response while connecting with websocket to rabbitmq

### ISSUE THREAD

**Issue by aevtyushkin** [shamblett/mqtt_client#530]
Websockets in RabbitMQ broker are implemented in rabbitmq_web_mqtt plugin.

Here is sample rabbitmq configuration (ssl peer verification is important one in my case):

```
[
    {rabbitmq_web_mqtt, [
        {ssl_config, [
            {ip,                   "0.0.0.0"},
            {port,                 443},
            {backlog,              1024},
            {cacertfile,           "/etc/rabbitmq/ssl/ca.crt"},
            {certfile,             "/etc/rabbitmq/ssl/srv.crt"},
            {keyfile,              "/etc/rabbitmq/ssl/srv.key"},
            {depth,                2},
            {verify,               verify_peer},
            {fail_if_no_peer_cert, true},
            {versions,             ['tlsv1.2']},
            {honor_cipher_order,   true},
            {honor_ecc_order,      true},
            {secure_renegotiate,   true}
        ]},
        {cowboy_opts, [
            {idle_timeout, 60000}
        ]},
        {ws_opts, [
            {idle_timeout, 60000}
        ]}
    ]}
].
```

Here is sample mqtt_client configuration

```
client = MqttServerClient("wss://example.com:443/ws", deviceId.value);
client.useWebSocket = true;
client.port = 443;
client.keepAlivePeriod = 50;
client.setProtocolV311();
client.secure = false;
client.logging(on: false);
client.websocketProtocols = ['mqtt'];
client.securityContext = securityContext;
client.useAlternateWebSocketImplementation = true;
```

I am using alternate WebSocket implementation since it has security context, so I could easily pass client certificate there.

So, now these combination brings to error 400 Bad Request from broker. 

Digging into library led to issue in websocket handshake process - there is a space in GET header right after HTTP/1.1 that mishandled by the broker.

I have added patch to fix this issue.

[rabbitmq.patch](https://github.com/shamblett/mqtt_client/files/14921699/rabbitmq.patch)

**Comment by shamblett** [comment]
OK thanks I'll update the packlage.

It doesnt surprise me there are bugs in the alternate web socket code, very few people actually use this.

**Comment by shamblett** [comment]
Package updated and re published at version 10.2.1

**Comment by aevtyushkin** [comment]
Everything works great. Thank you for this amazing project!
